exporter/elasticsearchexporter/model.go

```diff
@@ -64,8 +64,9 @@ var resourceAttrsToPreserve = map[string]bool{
 
 type mappingModel interface {
 	encodeLog(pcommon.Resource, plog.LogRecord, pcommon.InstrumentationScope) ([]byte, error)
-	encodeMetrics(pcommon.Resource, pmetric.MetricSlice, pcommon.InstrumentationScope) ([][]byte, error)
 	encodeSpan(pcommon.Resource, ptrace.Span, pcommon.InstrumentationScope) ([]byte, error)
+	upsertMetricDataPoint(map[uint32]objmodel.Document, pcommon.Resource, pcommon.InstrumentationScope, pmetric.Metric, pmetric.NumberDataPoint) error
```

Comment:
Yes, this is ugly. But the "grouping" idea doesn't fit into the separation between exporter and encoding model. Happy to take any suggestions. If it isn't blocking, we can defer it to a future refactoring PR.